src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -528,14 +528,19 @@ bool Ekf::checkAndFixCovarianceUpdate(const SquareMatrixState &KHP)
 void Ekf::resetQuatCov(const float yaw_noise)
 {
 	const float tilt_var = sq(fmaxf(_params.initial_tilt_err, 1.0e-2f));
-	Vector3f rot_var_ned(tilt_var, tilt_var, 0.f);
+	float yaw_var = 0.0f;
 
 	// update the yaw angle variance using the variance of the measurement
 	if (PX4_ISFINITE(yaw_noise)) {
 		// using magnetic heading tuning parameter
-		rot_var_ned(2) = (sq(fmaxf(yaw_noise, 1.0e-2f)));
+		yaw_var = (sq(fmaxf(yaw_noise, 1.0e-2f)));
 	}
 
+	resetQuatCov(Vector3f(tilt_var, tilt_var, yaw_var));
```

Comment:
Should we provide a lower bound here? If the caller doesn't provide a yaw_noise then we're calling resetQuatCov with 0 yaw variance.

Reply:
Yes definitely.

Reply:
Done